test/client/fixtures/sandbox/shadow-ui-test.js

```diff
@@ -835,3 +835,34 @@ test('getElementsByTagName', function () {
     strictEqual(elements[0], textarea1);
     strictEqual(elements[1], void 0);
 });
+
+test('live collection should properly change if element was added (gh-1111)', function () {
+    var testDiv         = $('#testDiv')[0];
+    var textarea1       = document.createElement('textarea');
+    var textarea2       = document.createElement('textarea');
+    var textarea3       = document.createElement('textarea');
+    var textarea4       = document.createElement('textarea');
+    var shadowUIElement = document.createElement('textarea');
+
+    shadowUI.addClass(shadowUIElement, 'el');
+
+    testDiv.appendChild(textarea1);
+    testDiv.appendChild(shadowUIElement);
+
+    var elements = document.getElementsByTagName('textarea');
+
+    strictEqual(elements.length, 1);
+    strictEqual(elements[0], textarea1);
+    strictEqual(elements[1], void 0);
+
+    testDiv.appendChild(textarea2);
+    testDiv.appendChild(textarea3);
```

Comment:
is there any reason to add more than one additional textarea here? As I see, one is enough